src/Stripe.net/Infrastructure/StringOrObject.cs

```diff
@@ -4,20 +4,21 @@ namespace Stripe.Infrastructure
     using Newtonsoft.Json.Linq;
 
     internal static class StringOrObject<T>
-        where T : StripeEntityWithId
+        where T : IStripeEntityWithId
     {
         public static void Map(object value, Action<string> updateId, Action<T> updateObject)
         {
             if (value is JObject)
             {
-                T item = ((JToken)value).ToObject<T>();
+                // T item = ((JToken)value).ToObject<T>();
+                T item = Mapper<T>.MapFromJson(value.ToString());
```

Comment:
Not fully sure what all of this does

Reply:
It uses our custom deserialization instead of the default one, to ensure that our registered converters are used.